x/staking/keeper/grpc_query.go

```diff
@@ -56,12 +57,25 @@ func (k Querier) Validators(ctx context.Context, req *types.QueryValidatorsReque
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	vals := types.Validators{}
+	var vals []types.ValidatorInfo
 	for _, val := range validators {
-		vals.Validators = append(vals.Validators, *val)
+		valInfo := types.ValidatorInfo{
+			Validator: *val,
+		}
+
+		cpk, ok := val.ConsensusPubkey.GetCachedValue().(cryptotypes.PubKey)
+		// Best-effort way
+		if ok {
+			consAddr, _ := k.consensusAddressCodec.BytesToString(sdk.ConsAddress(cpk.Address()))
```

Comment:
```suggestion
			consAddr, _ := k.consensusAddressCodec.BytesToString(cpk.Address())
```

nit: no need to use sdk.ConsAddress iirc